main.tf

```diff
@@ -70,7 +68,7 @@ resource "helm_release" "castai_cluster_controller" {
 
   set {
     name  = "castai.clusterID"
-    value = data.castai_eks_clusterid.castai_cluster_id.id
+    value =  castai_eks_cluster.my_castai_cluster.id
```

Comment:
Wait but if you do this, won't cluster-controller get deleted before castai_eks_cluster?

Reply:
We shouldn't have problems for disconnect flow if cluster-controller is deleted before cluster.
In disconnect api we have `KeepKubernetesResources` option. For terraform we could pass false and cleanup resource in terraform which will also be faster and more stable.

Reply:
`KeepKubernetesResources=true` is allowing us to cleanup resources in terraform, due to that flag mothership is not creating task for cluster-controller to cleanup Kubernetes resources